runtime/vm/bindnatv.cpp

```diff
@@ -653,6 +653,12 @@ nativeSignature(J9Method* nativeMethod, char *resultBuffer)
 				nextType = 'L';
 				while(J9UTF8_DATA(methodSig)[i++] != ';') {}		/* a type string - loop scanning for ';' to end it - i points past ';' when done loop */
 				break;
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			case 'Q':
+				nextType = 'Q';
+				while(J9UTF8_DATA(methodSig)[i++] != ';') {}		/* a type string - loop scanning for ';' to end it - i points past ';' when done loop */
+				break;
+#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
```

Comment:
This can be merged into the case for `L` by changing the assignment to `nextType = ch;`.